mmv1/third_party/terraform/services/compute/resource_compute_subnetwork_test.go.tmpl

```diff
@@ -152,6 +154,45 @@ func TestAccComputeSubnetwork_update(t *testing.T) {
 	}
 }
 
+func TestAccComputeSubnetwork_purposeUpdate(t *testing.T) {
+        t.Parallel()
+
+        var subnetwork compute.Subnetwork
+	cnName := fmt.Sprintf("tf-test-%s", acctest.RandString(t, 10))
+	subnetworkName := fmt.Sprintf("tf-test-%s", acctest.RandString(t, 10))
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckComputeSubnetworkDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				// Create a subnetwork with the purpose set to PEER_MIGRATION
+				Config: testAccComputeSubnetwork_purposeUpdate(cnName, subnetworkName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeSubnetworkExists(t, "google_compute_subnetwork.network-with-migration-purpose", &subnetwork),
+				),
+			},
```

Comment:
you should add an `ImportStateVerify` step between steps 1 & 2 to make sure that the API values are set properly in the previous step.

```suggestion
			},
			{
				ResourceName:      "google_compute_subnetwork.network-with-migration-purpose",
				ImportState:       true,
				ImportStateVerify: true,
			},
```